internal/store/builder.go

```diff
@@ -219,32 +219,41 @@ func (b *Builder) WithCustomResourceStoreFactories(fs ...customresource.Registry
 	}
 }
 
-// WithAllowAnnotations configures which annotations can be returned for metrics
-func (b *Builder) WithAllowAnnotations(annotations map[string][]string) {
-	if len(annotations) > 0 {
-		b.allowAnnotationsList = annotations
+func (b *Builder) allowLabels(labels map[string][]string) (map[string][]string, error) {
```

Comment:
Can you rename this function to `allowList` (if it's used for annotations and labels) or similar and add a descriptive comment on what it does?

Also please rename the labels argument to something more generic